tests/test_adm_yang.py

```diff
@@ -1104,3 +1106,104 @@ def test_var_params(self):
         self.assertEqual('var_with_param', obj.norm_name)
         self.assertEqual(1, len(obj.parameters.items))
 
+    def test_decode_rules(self):
+        buf = self._get_mod_buf('''
+  amm:sbr sbr1 {
+    amm:enum 8;
+    description
+      "";
+    amm:action "/AC/(./CTRL/first,./CTRL/second)";
+    amm:condition "/AC/(./EDD/sensor,./VAR/min_threshold,./OPER/compare_lt)";
+    amm:min-interval "/TD/PT30S";
+    amm:init-enabled false;
+    amm:max-count 10;
+  }
+
+  amm:sbr sbr2 {
+    amm:enum 9;
+    description
+      "";
+    amm:action "/AC/(./CTRL/first,./CTRL/second)";
+    amm:condition "/AC/(./EDD/sensor,./VAR/min_threshold,./OPER/compare_lt)";
+  }
+
+  amm:tbr tbr1 {
+    amm:enum 6;
+    description
+      "";
+    amm:action "/AC/(./CTRL/first,./CTRL/second)";
+    amm:period "/TD/PT30S";
+    amm:start "/TD/PT30S";
+    amm:init-enabled false;
+    amm:max-count 10;
+  }
+
+  amm:tbr tbr2 {
+    amm:enum 7;
+    description
+      "";
+    amm:action "/AC/(./CTRL/first,./CTRL/second)";
+    amm:period "/TD/PT30S";
+  }
+''')
+        with self.assertLogs(adm_yang.LOGGER, level=logging.WARNING) as logs:
+            adm = self._adm_dec.decode(buf)
+        self.assertEqual(
+            [],
+            self._filter_logs(logs.output)
+        )
+        self.assertIsInstance(adm, models.AdmModule)
+        self._db_sess.add(adm)
+        self._db_sess.commit()
+        self.assertIsNone(adm.source.abs_file_path)
+
+        self.assertEqual('example-mod', adm.module_name)
+        self.assertEqual('example', adm.ns_org_name)
+        self.assertEqual(65535, adm.ns_org_enum)
+        self.assertEqual('mod', adm.ns_model_name)
+        self.assertEqual(1, adm.ns_model_enum)
+        self.assertEqual(1, len(adm.imports))
+        self.assertEqual(1, len(adm.revisions))
+
+        self.assertEqual(2, len(adm.sbr))
+        obj = adm.sbr[0]
+        self.assertIsInstance(obj, models.Sbr)
+        self.assertEqual("sbr1", obj.name)
+
+        obj = adm.sbr[1]
+        self.assertIsInstance(obj, models.Sbr)
+        self.assertEqual("sbr2", obj.name)
+        self.assertEqual(2, len(adm.sbr))
+
+        self.assertEqual("/AC/(./CTRL/first,./CTRL/second)", adm.sbr[0].action_value)
```

Comment:
Does the decoded ARI get resolved upon reading from ADM? It makes sense that the text argument stays this way but is there a way to test the decoded references?

Reply:
Thanks Brian! At this point let me move your comment into a separate issue.